packages/@expo/config-plugins/src/utils/plugin-resolver.ts

```diff
@@ -76,6 +75,26 @@ export function moduleNameIsDirectFileReference(name: string): boolean {
   return slashCount > 1;
 }
 
+const exportsAwareSilentResolve = (projectRoot: string, pluginReference: string): string | null => {
+  try {
+    return resolveFrom(projectRoot, pluginReference);
+  } catch (error) {
+    if (
+      error &&
+      typeof error === 'object' &&
+      'code' in error &&
+      error.code === 'ERR_PACKAGE_PATH_NOT_EXPORTED'
+    ) {
+      throw new PluginError(
```

Comment:
would be ideal if this would just warn when the app.plugin.js is found but not reachable via package.json exports.

Reply:
The error does not mean "app.plugin.js is found but not reachable via package.json exports." it only means "app.plugin.js is not reachable via package.json exports.".

I have looked into detecting that the `app.plugin.js` file is available on the fs and using it, but I have mixed feelings about it.

We could detect the `package.json` location and derive the `app.plugin.js` location from that. But that relies on `package.json` being exported. Or we find the `main` entry with `require.resolve("test-lib-with-exports", {paths: [projectRoot]})`, which gives something like `expo/packages/@expo/config-plugins/src/utils/__tests__/fixtures/node_modules/test-lib-with-exports/lib/module/index.js` and then we'd need to walk up the the directory tree and look for the plugin file. Not great either.